### ISSUE TITLE
Can't apply CoreDisplay Patch (Mac OS 10.14.4)

### ISSUE THREAD

**Issue by sergiu-siminiuc** [Floris497/mac-pixel-clock-patch-V2#302]
I get this error:
SIP looks to be disabled, all good!

Unknown version of CoreDisplay found.
---- BEGINNING MD5 HASH SUMS ---- version: 10.14.4 build:18E226

     otool CoreDisplay: 176fcc0d47bcdfb463357c3864bf9bd1
 otool CoreDisplay.bak: NO FILE (this is okay)
           CoreDisplay: 0b7eb75c65f7a1d0620f0e0b62c939ea
       CoreDisplay.bak: NO FILE (this is okay)

---- ENDING MD5 HASH SUMS -------

**Comment by sergiu-siminiuc** [comment]
I'm using a macbook pro 13" 2017

**Comment by chinacoolboyzz** [comment]
```
CoreDisplay-patcher.command
oToolCoreDisplayUnpatched=(
  ...
  714a9e14ccc64b0cf4bef2f083087d8e '10.14.2 (18C54)' 5
  176fcc0d47bcdfb463357c3864bf9bd1 '10.14.4 (18E226)' 5
)

oToolCoreDisplayPatched=(
  ...
  2d71736504e14882b2eb5e994859ec09 '10.14.2 (18C54)'
  cb134eba367695eee1ba01c511121a41 '10.14.4 (18E226)'
)
```

**Comment by jeffjacobsen** [comment]
@chinacoolboyzz - That worked for me on a MacMini - Thanks!

**Comment by Slackboi** [comment]
@chinacoolboyzz patch之后外接屏无信号，mac这边却检测到了外接显示器

**Comment by Slackboi** [comment]
@chinacoolboyzz 

"
Password:
Re-signing /System/Library/Frameworks/CoreDisplay.framework/Versions/A/CoreDisplay
/System/Library/Frameworks/CoreDisplay.framework/Versions/A/CoreDisplay: replacing existing signature
update_dyld_shared_cache: warning: x86_64h skipping because of bad install name /System/Library/PrivateFrameworks/FaceCore.framework/Versions/A/Resources/fcl-fc-1.dat
update_dyld_shared_cache: warning: x86_64h skipping because of bad install name /System/Library/PrivateFrameworks/FaceCore.framework/Versions/A/Resources/fcl-fc-2.dat
update_dyld_shared_cache: warning: x86_64h skipping because of bad install name /System/Library/PrivateFrameworks/FaceCore.framework/Versions/A/Resources/fcl-fc-3.dat
update_dyld_shared_cache: warning: i386 skipping because of bad install name /System/Library/PrivateFrameworks/FaceCore.framework/Versions/A/Resources/fcl-fc-1.dat
update_dyld_shared_cache: warning: i386 skipping because of bad install name /System/Library/PrivateFrameworks/FaceCore.framework/Versions/A/Resources/fcl-fc-2.dat
update_dyld_shared_cache: warning: i386 skipping because of bad install name /System/Library/PrivateFrameworks/FaceCore.framework/Versions/A/Resources/fcl-fc-3.dat
update_dyld_shared_cache: i386 incorporating 629 OS dylibs, tracking 457 others, building closures for 247 executables
update_dyld_shared_cache: warning: x86_64h rejected from cached dylibs: /System/Library/PrivateFrameworks/CreateML.framework/Versions/A/CreateML (("Could not find dependency '/System/Library/PrivateFrameworks/TuriCore.framework/Versions/A/TuriCore'"))
update_dyld_shared_cache: warning: x86_64h rejected from cached dylibs: /usr/lib/swift/libswiftCreateML.dylib (("Could not find dependency '/System/Library/PrivateFrameworks/TuriCore.framework/Versions/A/TuriCore'"))
update_dyld_shared_cache: x86_64h incorporating 1383 OS dylibs, tracking 1004 others, building closures for 2137 executables
Closing...  "
你好!上面是我的控制台输出。我patch之后,外接显示器会黑屏。10.14.4

**Comment by Slackboi** [comment]
@chinacoolboyzz 你好，我的mbp retina 13在使用补丁之后外接显示器黑屏了，下面是我的控制台输出，可以请问这是怎么回事吗?谢谢

> ---- BEGINNING MD5 HASH SUMS ---- version: 10.14.4 build:18E226
> otool CoreDisplay: 176fcc0d47bcdfb463357c3864bf9bd1
> otool CoreDisplay.bak: NO FILE (this is okay)
> CoreDisplay: 0b7eb75c65f7a1d0620f0e0b62c939ea
> CoreDisplay.bak: NO FILE (this is okay)
> ---- ENDING MD5 HASH SUMS -------

> > ```
> > Password:
> > Re-signing /System/Library/Frameworks/CoreDisplay.framework/Versions/A/CoreDisplay
> > /System/Library/Frameworks/CoreDisplay.framework/Versions/A/CoreDisplay: replacing > existing signature
> > update_dyld_shared_cache: warning: x86_64h skipping because of bad install name > /System/Library/PrivateFrameworks/FaceCore.framework/Versions/A/Resources/fcl-fc-1.dat
> > update_dyld_shared_cache: warning: x86_64h skipping because of bad install name > /System/Library/PrivateFrameworks/FaceCore.framework/Versions/A/Resources/fcl-fc-2.dat
> > update_dyld_shared_cache: warning: x86_64h skipping because of bad install name > /System/Library/PrivateFrameworks/FaceCore.framework/Versions/A/Resources/fcl-fc-3.dat
> > update_dyld_shared_cache: warning: i386 skipping because of bad install name > /System/Library/PrivateFrameworks/FaceCore.framework/Versions/A/Resources/fcl-fc-1.dat
> > update_dyld_shared_cache: warning: i386 skipping because of bad install name > /System/Library/PrivateFrameworks/FaceCore.framework/Versions/A/Resources/fcl-fc-2.dat
> > update_dyld_shared_cache: warning: i386 skipping because of bad install name > /System/Library/PrivateFrameworks/FaceCore.framework/Versions/A/Resources/fcl-fc-3.dat
> > update_dyld_shared_cache: i386 incorporating 629 OS dylibs, tracking 457 others, building > closures for 247 executables
> > update_dyld_shared_cache: warning: x86_64h rejected from cached dylibs: 
> >  /System/Library/PrivateFrameworks/CreateML.framework/Versions/A/CreateML (("Could not > find dependency 
> >  '/System/Library/PrivateFrameworks/TuriCore.framework/Versions/A/TuriCore'"))
> > update_dyld_shared_cache: warning: x86_64h rejected from cached dylibs: 
> >  /usr/lib/swift/libswiftCreateML.dylib (("Could not find dependency 
> > '/System/Library/PrivateFrameworks/TuriCore.framework/Versions/A/TuriCore'"))
> > update_dyld_shared_cache: x86_64h incorporating 1383 OS dylibs, tracking 1004 others, > > building closures for 2137 executables
> > Closing...
> > ```

**Comment by Slackboi** [comment]
@chinacoolboyzz 你好，我的mbp retina 13在使用补丁之后外接显示器黑屏了，下面是我的控制台输出，可以请问这是怎么回事吗?谢谢

> ---- BEGINNING MD5 HASH SUMS ---- version: 10.14.4 build:18E226
> otool CoreDisplay: 176fcc0d47bcdfb463357c3864bf9bd1
> otool CoreDisplay.bak: NO FILE (this is okay)
> CoreDisplay: 0b7eb75c65f7a1d0620f0e0b62c939ea
> CoreDisplay.bak: NO FILE (this is okay)
> ---- ENDING MD5 HASH SUMS -------

> ```
> Password:
> Re-signing /System/Library/Frameworks/CoreDisplay.framework/Versions/A/CoreDisplay
> /System/Library/Frameworks/CoreDisplay.framework/Versions/A/CoreDisplay: replacing > existing signature
> update_dyld_shared_cache: warning: x86_64h skipping because of bad install name > /System/Library/PrivateFrameworks/FaceCore.framework/Versions/A/Resources/fcl-fc-1.dat
> update_dyld_shared_cache: warning: x86_64h skipping because of bad install name > /System/Library/PrivateFrameworks/FaceCore.framework/Versions/A/Resources/fcl-fc-2.dat
> update_dyld_shared_cache: warning: x86_64h skipping because of bad install name > /System/Library/PrivateFrameworks/FaceCore.framework/Versions/A/Resources/fcl-fc-3.dat
> update_dyld_shared_cache: warning: i386 skipping because of bad install name > /System/Library/PrivateFrameworks/FaceCore.framework/Versions/A/Resources/fcl-fc-1.dat
> update_dyld_shared_cache: warning: i386 skipping because of bad install name > /System/Library/PrivateFrameworks/FaceCore.framework/Versions/A/Resources/fcl-fc-2.dat
> update_dyld_shared_cache: warning: i386 skipping because of bad install name > /System/Library/PrivateFrameworks/FaceCore.framework/Versions/A/Resources/fcl-fc-3.dat
> > update_dyld_shared_cache: i386 incorporating 629 OS dylibs, tracking 457 others, building > closures for 247 executables
> > update_dyld_shared_cache: warning: x86_64h rejected from cached dylibs: 
> >  /System/Library/PrivateFrameworks/CreateML.framework/Versions/A/CreateML (("Could not > find dependency 
>  '/System/Library/PrivateFrameworks/TuriCore.framework/Versions/A/TuriCore'"))
> update_dyld_shared_cache: warning: x86_64h rejected from cached dylibs: 
>  /usr/lib/swift/libswiftCreateML.dylib (("Could not find dependency 
> '/System/Library/PrivateFrameworks/TuriCore.framework/Versions/A/TuriCore'"))
> update_dyld_shared_cache: x86_64h incorporating 1383 OS dylibs, tracking 1004 others, > > building closures for 2137 executables
> Closing...
> ```

**Comment by chinacoolboyzz** [comment]
@Slackboi 
运行下 CoreDisplay-patcher.command md5
看你生成的 md5  是否等于 cb134eba367695eee1ba01c511121a41

**Comment by Slackboi** [comment]
> @Slackboi
> 运行下 CoreDisplay-patcher.command md5
> 看你生成的 md5 是否等于 cb134eba367695eee1ba01c511121a41
> 
> ```
> ./CoreDisplay-patcher.command md5                                                                                                                                                                                                                                                           ✔  0.02 Dur  3.73 L
> ---- BEGINNING MD5 HASH SUMS ---- version: 10.14.4 build:18E226
> 
>      otool CoreDisplay: cb134eba367695eee1ba01c511121a41
>  otool CoreDisplay.bak: 176fcc0d47bcdfb463357c3864bf9bd1
>            CoreDisplay: 0fe4ff987a2df4ebdb83bf37060210d3
>        CoreDisplay.bak: 0b7eb75c65f7a1d0620f0e0b62c939ea
> 
> ---- ENDING MD5 HASH SUMS -------
> ```

你好这是我生成的md5
>  ---- BEGINNING MD5 HASH SUMS ---- version: 10.14.4 build:18E226
>     otool CoreDisplay: 176fcc0d47bcdfb463357c3864bf9bd1
> otool CoreDisplay.bak: NO FILE (this is okay)
>           CoreDisplay: 0b7eb75c65f7a1d0620f0e0b62c939ea
>       CoreDisplay.bak: NO FILE (this is okay)
> ---- ENDING MD5 HASH SUMS -------

**Comment by kuhnskc** [comment]
> ```
> CoreDisplay-patcher.command
> oToolCoreDisplayUnpatched=(
>   ...
>   714a9e14ccc64b0cf4bef2f083087d8e '10.14.2 (18C54)' 5
>   176fcc0d47bcdfb463357c3864bf9bd1 '10.14.4 (18E226)' 5
> )
> 
> oToolCoreDisplayPatched=(
>   ...
>   2d71736504e14882b2eb5e994859ec09 '10.14.2 (18C54)'
>   cb134eba367695eee1ba01c511121a41 '10.14.4 (18E226)'
> )
> ```

I added this in and ran it. It appears to have worked but when I reboot, I'm still stuck on 30Hz. Any ideas?

10.14.4 (18E226)
MacBook Pro (13-inch, 2016, Two Thunderbolt 3 ports)
2 GHz Intel Core i5
8 GB 1867 MHz LPDDR3
Intel Iris Graphics 540 1536 MB

kcs-macbook-pro:~ kckuhns$ /Users/kckuhns/Desktop/CoreDisplay-patcher.command status
SIP looks to be disabled, all good!

(otool) Detected patched CoreDisplay on OS X 10.14.4 (18E226).

**Comment by 747857485** [comment]
@chinacoolboyzz  大神我的是 10.14.6 系统，提示：
If you are running a new OS, run ./CoreDisplay-patcher.command md5 and ask the maintainer of this script to add support for your system.

我的md5 是这样的
     otool CoreDisplay: 105051509e563ba96f190ed78a52feb9
 otool CoreDisplay.bak: 105051509e563ba96f190ed78a52feb9
           CoreDisplay: 7ff72e56ac57c57b3a347561497f60d6
       CoreDisplay.bak: 11d7f86a932ae39d3a248abc40a68b43

请问command 文件应该怎么修改？
